### PR TITLE
streamlined SingularityBWASP to also allow BWASPR execution

### PR DESCRIPTION
**Pull request by vpbrendel**
Steps towards full implementation of singularity:
1) new recipe
2) simplified makefile_pe_template

To do:
1) add makefile_se_template and makefile_rrbs_template
2) re-write of merge makefiles
3) BrendelGroup submission of finished singularity container
4) updates to how to build a directory structure for complete reproduction via "xdoit"

### CODE REVIEW

**Comment by littleblackfish** [comment]
1. Recipe file needs to be named 'Singularity' for the singularity hub hooks to work. Just like makefiles, we should follow convention here: 1 recipe per repository, and that should be called 'Singularity'. 
2. You should go to https://singularity-hub.org/ and register for BrendelGroup and connect the repositories so that our containers will actually be built. Then we can do `singularity pull BrendelGroup/BWASP` which looks more appropriate than littleblackfish.
3. Second `apt -y update` in recipe looks redundant, how is that different than `apt-get -y update` ?
4. `apt -y full-upgrade` is probably not the best idea for a container. We want to work with versions we know, not the bleeding edge. Up to debate. 
5. makefile_pe_template replaces Makefile_WF1-6pe_template ? I agree that is a much better name. 
6. If we are going to install all dependencies in the beginning we should annotate the non-standard ones at least. Why do we need libssl-dev, for example? Or software-properties-common? I tried to keep mine clustered with their dependents, although that was also imperfect. Recipe should not be too hard to decipher. Up to debate.  
7. Do we really need JDK, and not JRE ? Let's keep it lean if we can.

**Comment by vpbrendel** [comment]
1,2) Will be taken care of when a bit more debugging is done.
3,4) I'll think about that.  If we test the container before we build it on the hub, then a full upgrade would be ok; but we may not need it.
5) yes
6,7) These are dependencies to make the R packages install. It's too awkward to list all the precise dependencies.  If someone does not like it, then can delete something and find out what fails downstream ...

File: README.md [hunk 1]
```diff
@@ -9,8 +9,8 @@ and __easy__ to use.
 
 BWASP is built on various open source software that was developed within our group and elsewhere.
 To achieve a portable and reproducible workflow, we encapsulated all the dependencies in a [Singularity](http://singularity.lbl.gov) container.
-The actual workflows are implemented in simple makefiles that refer to this self-contained environment.
-As such, __the only requirement for BWASP is Singularity__ itself.
+The workflow is implemented with GNU make.
+If the provide makefile is invoked via the proper singularity command using our BWASP container, no further installation of programs in necessary.
```
Comment (littleblackfish) [inline README.md]: typo : provide-> provided.

**Comment by littleblackfish** [comment]
I have no major complaints other than minor typo in README that happened to strike my attention. Should we accept this request and then run the actual test, or should I test with your repo first and then accept the pull?

**Pull request merged**